src/System.ComponentModel.Annotations/tests/ValidationAttributeTestBase.cs

```diff
@@ -0,0 +1,135 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace System.ComponentModel.DataAnnotations.Tests
+{
+    public abstract class ValidationAttributeTestBase
+    {
+        public abstract IEnumerable<Test> ValidValues();
+        public abstract IEnumerable<Test> InvalidValues();
```

Comment:
It's too bad xunit doesn't currently enable a mechanism for a base class to define a MemberData that looks at a derived type.  I guess this is the best we can do right now.